db/template_test.go

```diff
@@ -364,6 +431,137 @@ func TestGetTemplateWithInvalidID(t *testing.T) {
 	}
 }
 
+func TestUpdateTemplate(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+	_, tinkDB, cl := NewPostgresDatabaseClient(t, ctx, NewPostgresDatabaseRequest{
+		ApplyMigration: true,
+	})
+	defer func() {
+		err := cl()
+		if err != nil {
+			t.Error(err)
+		}
+	}()
+
+	// create template
+	w := workflow.MustParseFromFile("./testdata/template_happy_path_1.yaml")
+	w.ID = uuid.New().String()
+	w.Name = fmt.Sprintf("id_%d", rand.Int())
+	err := createTemplateFromWorkflowType(ctx, tinkDB, w)
+	if err != nil {
+		t.Error(err)
+	}
+
+	// validate revisions
+	revisions := 0
+	err = tinkDB.ListTemplateRevisions(w.ID, func(id string, revision int, tCr *timestamp.Timestamp) error {
+		revisions = revisions + 1
+		return nil
+	})
+	if err != nil {
+		t.Error(err)
+	}
+	if revisions != 1 {
+		t.Errorf("expected 1 template revisions stored in the database but we got %d", revisions)
+	}
+
+	// update template
+	w.Name = "updated-template-name"
+	data, err := json.Marshal(w)
+	if err != nil {
+		t.Error(err)
+	}
+	err = tinkDB.UpdateTemplate(ctx, w.Name, string(data), uuid.MustParse(w.ID))
+	if err != nil {
+		t.Error(err)
+	}
+
+	// revalidate revisions
+	revisions = 0
+	err = tinkDB.ListTemplateRevisions(w.ID, func(id string, revision int, tCr *timestamp.Timestamp) error {
+		revisions = revisions + 1
+		return nil
+	})
+	if err != nil {
+		t.Error(err)
+	}
+	if revisions != 2 {
+		t.Errorf("expected 2 template revisions stored in the database but we got %d", revisions)
+	}
+}
+
+func TestTemplateRevisionMigration(t *testing.T) {
+	t.Parallel()
+	ctx := context.Background()
+	_, tinkDB, cl := NewPostgresDatabaseClient(t, ctx, NewPostgresDatabaseRequest{})
+	defer func() {
+		err := cl()
+		if err != nil {
+			t.Error(err)
+		}
+	}()
+
+	err := tinkDB.WithMigrations([]func() *migrate.Migration{
+		migration.Get202009171251,
+		migration.Get202010071530,
+		migration.Get202010221010,
+		migration.Get202012041103,
+		migration.Get202012091055,
+		migration.Get2020121691335,
+	})
+	if err != nil {
+		t.Error(err)
+	}
+
+	// create a few templates
+	w := workflow.MustParseFromFile("./testdata/template_happy_path_1.yaml")
+	for i := 1; i <= 5; i++ {
+		id := uuid.New().String()
+		w.ID = id
+		w.Name = fmt.Sprintf("id_%d", rand.Int())
+		err := createTemplateFromWorkflowType(ctx, tinkDB, w)
```

Comment:
This test fails because the function `CreateTemplate` already relies on revision. Even if you didn't apply the migration yet, the Go code assumes revision to be in place.

In order to fix that you have to add a template not via code but with a SQL query.